src/NServiceBus.Persistence.ServiceFabric/SagaPersister/RuntimeSagaInfo.cs

```diff
@@ -33,8 +33,8 @@ class RuntimeSagaInfo
 
             SagaAttribute = new ServiceFabricSagaAttribute
             {
-                CollectionName = !string.IsNullOrEmpty(userProvidedAttribute?.CollectionName) ? userProvidedAttribute.CollectionName : sagaDataType.Name,
-                SagaDataName = !string.IsNullOrEmpty(userProvidedAttribute?.SagaDataName) ? userProvidedAttribute.SagaDataName : sagaDataType.Name
+                CollectionName = userProvidedAttribute?.CollectionName ?? sagaDataType.Name,
```

Comment:
As we discussed, this potentially would fail during development as the key in a reliable collection cannot be an empty string. So we're good.